src/App.tsx

```diff
@@ -1,61 +1,158 @@
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
-    </div>
-  );
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+import users from './api/users';
+
+import { getUser } from './helpers/getUser';
+import { todos } from './helpers/getTodos';
+
+import { TodoList } from './components/TodoList';
+
+type State = {
+  currentTodo: Todo,
+  todos: Todo[],
+  re: RegExp,
+  inputStatusMonitoring: boolean,
 };
+
+export class App extends React.Component<{}, State> {
+  state = {
+    currentTodo: {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      title: '',
+      userId: 0,
+      completed: false,
+    },
+    todos: [...todos],
+    re: /^[A-Za-zА-Яа-я0-9 ]*$/i,
+    inputStatusMonitoring: false,
+  };
+
+  handleChange = (event:
+  React.FormEvent<HTMLInputElement>
+  | React.ChangeEvent<HTMLSelectElement>) => {
+    const { name, value } = event.target as HTMLInputElement;
+
+    if (value.match(this.state.re)) {
+      this.setState((prevState => ({
+        currentTodo: {
+          ...prevState.currentTodo,
+          [name]: value,
+        },
+      })));
+    }
+  };
+
+  handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (this.state.currentTodo.userId && this.state.currentTodo.title.trim()) {
+      this.setState(prevState => {
+        const newTodo = {
+          ...this.state.currentTodo,
+          user: getUser(this.state.currentTodo.userId),
+        };
+
+        return {
+          ...prevState,
+          currentTodo: {
+            ...prevState.currentTodo,
+            id: prevState.currentTodo.id + 1,
+          },
+          todos: [
+            ...prevState.todos,
+            newTodo,
+          ],
+        };
+      });
+
+      this.handleResetForm();
+    } else {
+      this.state.inputStatusMonitoring = true;
+
+      this.forceUpdate();
```

Comment:
Do we need `forceUpdate` here? You should avoid using this: https://reactjs.org/docs/react-component.html#forceupdate

So try to rewrite avoiding this